src/classes/CW_CampaignRewardCompControllerTest.cls

```diff
@@ -0,0 +1,260 @@
+		insert rewardCat;
+		
+		CW_Reward_Library__c rewardLib= TestDataUtil.createRewardLibrary(rewardType.Id);
+		insert rewardLib;
+		
+		CW_Campaign_Offer__c campOffer = TestDataUtil.createCampaignOffer(camp.Id, offerType.Id);
+		insert campOffer; 
+		
+		List<CW_Campaign_Reward__c> listRewards = new List<CW_Campaign_Reward__c>();
+		for (Integer i=0;i<11;i++) 
+		{
+			listRewards.Add(TestDataUtil.createCampaignReward(campOffer.Id, rewardCat.Id, rewardLib.Id, rewardType.Id));
+		}
+		insert listRewards;
+			
+    	CW_CampaignRewardCompController compCtrl= new CW_CampaignRewardCompController();
+    	compCtrl.campOfferId = campOffer.Id;
+    	compCtrl.campOffer = campOffer;
+    	compCtrl.offerTypeId = offerType.Id;
+    	compCtrl.init();
+    	compCtrl.getHasPrevious();
+    	compCtrl.getHasNext();
+    	compCtrl.getPageNumber();
+    	compCtrl.getTotalPages();
+    	compCtrl.getPageNumber();
+    	compCtrl.campId=camp.Id;
+
+		system.assertEquals(camp.Id	,compCtrl.campId);
+		system.assertEquals(10		,compCtrl.getRewardList().Size());
+		system.assertEquals(false  	,compCtrl.editMode);
+		system.assertEquals(false 	,compCtrl.disabledButtons);
+		system.assertEquals(true  	,compCtrl.getDeleteDisabled());
+		system.assertEquals(2     	,compCtrl.getTotalPages());
+		system.assertEquals(false 	,compCtrl.NewRecMode);
+		system.assertEquals(false 	,compCtrl.editMode);
+			
+		compCtrl.selectAll = true;
+		compCtrl.selectDeselectAllItems();
+		system.assertEquals(10     ,compCtrl.getSelectedCount());
+		system.assertEquals(false ,compCtrl.getDeleteDisabled());
+			
+		compCtrl.selectAll = false;
+		compCtrl.selectDeselectAllItems();
+		system.assertEquals(0     ,compCtrl.getSelectedCount());
+		system.assertEquals(true  ,compCtrl.getDeleteDisabled());
+			
+		compCtrl.contextItem = listRewards[2].Id;
+		compCtrl.doSelectItem();
+		system.assertEquals(1     ,compCtrl.getSelectedCount());
+		compCtrl.doDeselectItem();
+		system.assertEquals(0     ,compCtrl.getSelectedCount());
+			
+		compCtrl.last();
+		system.assertEquals(1     ,compCtrl.getRewardList().Size());
+			
+		compCtrl.doNext();
+		system.assertEquals(2     ,compCtrl.getPageNumber());
+		system.assertEquals(true  ,compCtrl.getHasPrevious());
+		system.assertEquals(false ,compCtrl.getHasNext());
+			
+		compCtrl.doPrevious();
+		system.assertEquals(false  ,compCtrl.getHasPrevious());
+		system.assertEquals(true   ,compCtrl.getHasNext());
+		
+		compCtrl.selectedRewardIds.Add(listRewards[3].Id);
+		compCtrl.selectedRewardIds.Add(listRewards[4].Id);
+		system.assertEquals(false  ,compCtrl.getDeleteDisabled());
+		compCtrl.getDeleteMessage();
+		compCtrl.deleteRewards();
+
+		system.assertEquals(9     ,compCtrl.getRewardList().Size());
+		system.assertEquals(false  ,compCtrl.editMode);
+
+		compCtrl.savedByParent =true;
+		system.assertEquals(false  ,compCtrl.editMode);
+		compCtrl.Cancel();
+		system.assertEquals(false  ,compCtrl.editMode);
+	}
+	
+	static testMethod void createCampaignReward() 
```

Comment:
Style issue: test methods should be annotated with @IsTest for better readibility